packages/components/inputs/rich-text-input/src/editor.types.ts

```diff
@@ -0,0 +1,111 @@
+import type { ReactNode } from 'react';
+
+type TNodes = {
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TDocument = {
+  getClosest: (
+    block: { key: unknown },
+    closest: (parent: { type: string }) => boolean
+  ) => boolean;
+  getParent?: (parentParam: string) => {
+    type: string;
+  };
+  text?: string;
+  nodes?: TNodes;
+};
+
+type TBlocks = {
+  size?: number;
+  some: (block: unknown) => boolean | void;
+  first: () => {
+    key: string;
+    type: string;
+  };
+  map: (node: ReactNode) => {
+    toArray: () => string[];
+  };
+};
+
+type TValue = {
+  blocks: TBlocks;
+  document: TDocument;
+  selection: {
+    isFocused: boolean;
+  };
+  activeMarks:
+    | {
+        some: (ascertMarkType: (mark: TMark) => boolean) => void;
+      }
+    | TMark[];
+  data: TData;
+};
+
+type TMark = {
+  type: string;
+};
+
+type TData = {
+  get: (key: string) => TSize;
+};
+
+type TSize = {
+  size: number;
+};
+
+type TWrapBlock = {
+  wrapBlock: (blockTag: string) => void;
+};
+
+type TNode = {
+  texts: (nodeParam?: Record<string, string>) => TNodeTextsResult[];
+};
+
+type TNodeTextsResult = [{ key: string; text: string }, string];
+
+export type TEditor = {
```

Comment:
I feel it will be better to expose `TEditor` and some of the other types in this file from  `rich-text-utils` instead of duplicating the types here.

Reply:
Having in mind it's only a temporary solution and I took advantage of @emmenko's [freedom of choice](https://github.com/commercetools/ui-kit/pull/2129#discussion_r814594227) 😅

Reply:
I understand @Rhotimee concern but I also agree this is a good solution as well.

If we wanted to use `rich-test-utils` types we would need to modify it to expose them, but then we would have to remove that exposure when the [`react-slate` update](https://github.com/commercetools/ui-kit/issues/2120) is finished.

So I think this time a little bit of duplication is a better choice.

Reply:
Good point about removing them again. Ok let's just keep them duplicated for now.